docker/Dockerfile.spdk

```diff
@@ -0,0 +1,22 @@
+FROM fedora:33 AS base
```

Comment:
Any reason to pin specifically Fedora 33 which is almost two years old and not a newer release?

Reply:
Switched to _fedora:36_

Reply:
fedora:36 provides pacific (16.2.x) packages, while currently we need quincy (17.2.x) or newer. See the issue https://github.com/ceph/ceph-nvmeof/issues/44 for details.

Reply:
BTW I'm currently testing to build `spdk/spdk` on CentOS Stream 8 and at least the build succeeds (I tried with CentOS Stream 9 but they changed the name of the devtools repo, so it requires some extra fixes).

I'm just saying this because CentOS (or any other "kinda" LTS distro) gives us some extra stability/room. CentOS Stream 8 EoL is in May 2024, while Fedora 36 is only 6 month away from EoL. CentOS Stream 9 EoL is ~2027.

Reply:
> I'm just saying this because CentOS (or any other "kinda" LTS distro) gives us some extra stability/room.

+1, let's pick an LTS distro.  As discussed elsewhere, we aren't going to be using provided Ceph packages, so them being potentially older is not an issue.